### PR TITLE
added scan function to tensorflow frontend

### LINKED ISSUE #11577: ivy.<scan>
**Why should this be implemented?**
Because 3+ of the native frameworks have it, and it's a complex function.

**Links to native framework implementations**
- [Jax](<FUNC_URL_DOCS>)
- [PyTorch](<FUNC_URL_DOCS>)
- [TensorFlow](<FUNC_URL_DOCS>)
- [NumPy](<FUNC_URL_DOCS>)

### PR DESCRIPTION
**Pull request by maher-mohsen**
Close #11577

### CODE REVIEW

**Comment by maher-mohsen** [comment]
@zaeemansari70 check this new PR please
Thanks

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by suvadityamuk** [comment]
Hi, @maher-mohsen!

Apologies for the delay, been a bit swamped actually.
Thank you so much for your contribution to Ivy! :grin: 

A few things you may have to do before this PR gets ready for merging are as follows:

1) It seems that you have not included a `fn` argument to your function definition, which does not match what TensorFlow implements for [`scan`](https://www.tensorflow.org/api_docs/python/tf/scan). Do you mind adding/making use of this parameter in your definition as well? :slightly_smiling_face: 

2) Your implementation depends on `ivy.scan()`, but it seems that such a function does not exist in the `ivy` namespace! In such a case, the three possible solutions are to make the backends yourself as a separate contribution, wait until someone else has done so, or make the function as a composition of other, already-existing functions of Ivy. If you believe you can follow any of those methods to implement, please feel free to mention the same over here!

3) It seems that your linting is off, do you mind making use of the [pre-commit hook](https://lets-unify.ai/ivy/contributing/setting_up.html#pre-commit) to make sure that all your files pass `black` and `flake8` tests?

4) It seems that your forked repository has not synced up with the current `master` branch. Do you mind doing a quick Merge from Master commit as well so that things are up to date?

Thank you so much, again! :slightly_smiling_face: Feel free to ask more any questions!

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
